core/utils/eth_client_utils.go

```diff
@@ -10,11 +12,14 @@ import (
 	retry "github.com/yetanotherco/aligned_layer/core"
 )
 
-func WaitForTransactionReceipt(client eth.InstrumentedClient, ctx context.Context, txHash gethcommon.Hash) (*types.Receipt, error) {
+// WaitForTransactionReceiptRetryable repeatedly attempts to fetch the transaction receipt for a given transaction hash.
+// If the receipt is not found, the function will retry with exponential backoff until the specified `waitTimeout` duration is reached.
+// If the receipt is still unavailable after `waitTimeout`, it will return an error.
+func WaitForTransactionReceiptRetryable(client eth.InstrumentedClient, ctx context.Context, txHash gethcommon.Hash, waitTimeout time.Duration) (*types.Receipt, error) {
 	receipt_func := func() (*types.Receipt, error) {
 		return client.TransactionReceipt(ctx, txHash)
 	}
-	return retry.RetryWithData(receipt_func, retry.MinDelay, retry.RetryFactor, retry.NumRetries, retry.MaxInterval, retry.MaxElapsedTime)
+	return retry.RetryWithData(receipt_func, retry.MinDelay, retry.RetryFactor, 0, retry.MaxInterval, waitTimeout)
```

Comment:
We should adjust these parameters. With the current configuration, we are retrying at intervals of `1 sec - 2 secs - 4 secs - 8 secs - 16 secs` and then stop retrying when the `waitTimeout` is reached. If we start retrying at the beginning of a block, we won't be able to retry in the third block due to the exponential backoff. I suggest using a `MaxInterval` of 2 secs (to retry every 2 seconds) and increment the `waitTimeout` a bit to ensure we can retry in the third block as well.

Reply:
You are right, good one!

Reply:
Solved in [475125d](https://github.com/yetanotherco/aligned_layer/pull/1286/commits/475125db9e982b3e220a055c0ffe29b3d76dc8db).